smarts/sstudio/types.py

```diff
@@ -501,6 +501,37 @@ def __hash__(self):
     def __eq__(self, other):
         return self.__class__ == other.__class__ and hash(self) == hash(other)
 
+@dataclass(frozen=True)
+class Trip:
+    """A route with a single actor type with name and unique id."""
+    def __init__(
+        self,
+        vehicle_name: str,
+        actor: TrafficActor,
+        route: Union[RandomRoute, Route],
+        depart: float = 0,
+    ):
+        self.vehicle_name = vehicle_name
+        self.actor = TrafficActor(self.vehicle_name)
+        self.route = route
+        """The route for the actor to attempt to follow."""
+        self.depart = depart
+        """Start time in seconds."""
+        # XXX: Defaults to 1 hour of traffic. We may want to change this to be "continual
+        #      traffic", effectively an infinite end.
+
+    @property
+    def id(self) -> str:
+        """The unique id of this trip."""
+        return self.vehicle_name
+
+    def __hash__(self):
+        # Custom hash since self.actors is not hashable, here we first convert to a
+        # frozenset.
+        return _pickle_hash((self.route, self.actor))
+
+    def __eq__(self, other):
+        return self.__class__ == other.__class__ and hash(self) == hash(other)
```

Comment:
Vehicle name can no longer be assigned in this case in Trip. Considering modifying this dataclass into standard format without __init__() and using __post_init__() to assign the actor.